### ISSUE TITLE
nmp installation problem

### ISSUE THREAD

**Issue by kittugs143** [trufflesuite/truffle#4658]
- [ ] I've [opened a support ticket](https://trufflesuite.zendesk.com/hc/en-us/requests/new) before filing this issue.

---------------------------

## Issue

Hi I cannot install "npm install -g truffle" on my computer I am getting so many warnings and deprecated issues.
Please help me how can i overcome this issue.

OS: Window 11

**Comment by gnidan** [comment]
Hi @kittugs143 could you provide more information so we could help? Also note that you may find more luck getting help if you reach out to our [official support channel](https://trufflesuite.zendesk.com/hc/en-us/requests/new) first, since our support team is excellent and handles this sort of thing for a living (whereas we here on the dev team do not). Happy to help, of course! But we'll definitely need to understand more about how you installed Node/NPM and what versions you are using before we can advise.

To start, what errors did you get? You say you got warnings, but it's possible that Truffle installed just fine. Could you try running `truffle version` to see if you have it installed? Thanks!

**Comment by ImmuneGit** [comment]
I also have a problem with update truffle from 5.4.29 to 5.4.30. 
My current environment: 
MacOS Catalina v10.15.7;
xcode-select version 2373;
Truffle v5.4.29 (core: 5.4.29)
Solidity v0.5.16 (solc-js)
Node v17.4.0
Web3.js v1.5.3

I've got this during "npm -g install truffle@5.4.30":
<img width="1139" alt="Screen Shot 2022-01-28 at 00 42 19" src="https://user-images.githubusercontent.com/25203571/151449715-2eb5fd62-11d3-4be6-8920-426aa7f48b86.png">

**Comment by kittugs143** [comment]
> Hi @kittugs143 could you provide more information so we could help? Also note that you may find more luck getting help if you reach out to our [official support channel](https://trufflesuite.zendesk.com/hc/en-us/requests/new) first, since our support team is excellent and handles this sort of thing for a living (whereas we here on the dev team do not). Happy to help, of course! But we'll definitely need to understand more about how you installed Node/NPM and what versions you are using before we can advise.
> 
> To start, what errors did you get? You say you got warnings, but it's possible that Truffle installed just fine. Could you try running `truffle version` to see if you have it installed? Thanks!

I followed your instructions and below is the result 

Truffle v5.4.30 (core: 5.4.30)
Solidity v0.5.16 (solc-js)
Node v16.13.2
Web3.js v1.5.3

But still i am unable to install truffle

**Comment by eggplantzzz** [comment]
Note that you should use npm version 6. We've had problems with versions greater than that.

**Comment by 777Rob** [comment]
> I also have a problem with update truffle from 5.4.29 to 5.4.30. My current environment: MacOS Catalina v10.15.7; xcode-select version 2373; Truffle v5.4.29 (core: 5.4.29) Solidity v0.5.16 (solc-js) Node v17.4.0 Web3.js v1.5.3
> 
> I've got this during "npm -g install truffle@5.4.30": <img alt="Screen Shot 2022-01-28 at 00 42 19" width="1139" src="https://user-images.githubusercontent.com/25203571/151449715-2eb5fd62-11d3-4be6-8920-426aa7f48b86.png">

I had same problem uninstalling truffle and installing version v5.3.14 fixed it

**Comment by ImmuneGit** [comment]
It seems they broke something because till 5.4.29 everything was fine. It happens with 5.4.30 and 5.4.31 (the last version) only.

**Comment by alephart** [comment]
True, if i try with version `npm i -g truffle@5.4.29`, it install (with many deprecated). But, what if I want the latest version... I have faith in truffle, but it lacks!

**Comment by eggplantzzz** [comment]
Hey @ImmuneGit and @alephart, I cannot reproduce your error. I have no problem installing the latest Truffle on my machine. Can you tell me what version of Node and npm you are using?

**Comment by eggplantzzz** [comment]
Try downgrading npm to version 6. Truffle has problems with versions > 7.

**Comment by juancpulidos-mds** [comment]
@eggplantzzz, this is my environment:
MacOS Catalina v10.15.7;
xcode-select version 2373;
node v14.16.0;
npm v8.3.0

**Comment by ImmuneGit** [comment]
@eggplantzzz Anyway, it doesn't explain why up to version 5.4.29 the package has been installed without troubles with Node v.17 and Npm v8. And suddenly it requires a downgrade. It points to the broken updates.

**Comment by dk8996** [comment]
Having the same. Added the following to the [Dockerfile](https://github.com/dk8996/starter-kit-docker/blob/main/truffle-suite/Dockerfile) for a clean install.  
`RUN npm install -g node-gyp
RUN npm install -g node-gyp-build`

Outside of Docker. 
`npm install -g node-gyp
npm install -g node-gyp-build`

**Comment by eggplantzzz** [comment]
I really have no idea why it worked up to that version, that is a good data point! However, Truffle has not yet officially supported those versions. Eventually we'll have to resolve this in the future and we have efforts to do so in the works right now.

**Comment by eggplantzzz** [comment]
I can also verify that I can install versions prior to 5.4.30 with Node 16 and npm 8.x.

**Comment by dk8996** [comment]
Just to give an update. I got it working without installing `node-gyp` and `node-gyp-build`. Clean install via Docker, using Node 17. Here is the [Dockerfile](https://github.com/dk8996/starter-kit-docker/blob/main/truffle-suite/Dockerfile). I hope this helps someone but one things I've learned, is to leverage Docker to build the dev environments.
